tests/utils/test_helpers.py

```diff
@@ -62,3 +67,11 @@ def test_parse_requires():
 )
 def test_utils_helpers_canonical_names(raw):
     assert canonicalize_name(raw) == "a-b-c"
+
+
+def test_utils_helpers_temporary_directory_readonly_file():
```

Comment:
Here a check whether the folder is removed successfully is missing, is it?

Reply:
The bug itself would throw `PermissionError` on the removal attempt. So in that regard the test shows we no longer run into that issue.
That being said checking that it _actually_ was removed is probably also a good idea 😄

Reply:
Alrighty, added some asserts. 👍